atlasdb-commons/src/main/java/com/palantir/util/AtlasDbMetrics.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import com.codahale.metrics.Slf4jReporter;
+
+public class AtlasDbMetrics {
```

Comment:
I think we'll actually want this in the `atlasdb-client` module, and we should simplify this as just a holder for the `MetricRegistry` and we should rely on the user handling all of the reporting and logging of metrics outside of AtlasDB. Basically I envisage just adding the following first, then we can wrap things like Transaction

```
package com.palantir.atlasdb.metrics;

import com.codahale.metrics.MetricRegistry;
import com.google.common.base.Preconditions;

public final class AtlasDbMetrics {

    private static volatile MetricRegistry metrics;

    private AtlasDbMetrics() { }

    public static void setMetrics(MetricRegistry metricRegistry) {
        Preconditions.checkNotNull(metricRegistry, "metric registry cannot be null");
        AtlasDbMetrics.metrics = metricRegistry;
    }

    public static MetricRegistry metrics() {
        MetricRegistry metricRegistry = AtlasDbMetrics.metrics; // single volatile read
        Preconditions.checkState(metricRegistry != null, "metric registry has not been initialized");
        return metricRegistry;
    }

}
```